autopilot/contractor/contractor.go

```diff
@@ -863,12 +869,16 @@ LOOP:
 		}
 
 		// check if we already have a contract with a host on that subnet
-		if shouldFilter && ipFilter.IsRedundantIP(host.NetAddress, host.PublicKey) {
+		if ctx.ShouldFilterRedundantIPs() && filter.IsRedundantIP(host) {
```

Comment:
Might want to rename the method to `HasRedundantIP` since we pass the host now instead of an address.